java/maven.indexer/src/org/netbeans/modules/maven/indexer/MinimalArtifactInfoRemoteIndexCreator.java

```diff
@@ -0,0 +1,204 @@
+        String info = ArtifactInfo.nvl(ai.getPackaging())
+                + ArtifactInfo.FS
+                + ai.getLastModified()
+                + ArtifactInfo.FS
+                + ai.getSize()
+                + ArtifactInfo.FS
+                + ai.getSourcesExists().toString()
+                + ArtifactInfo.FS
+                + ai.getJavadocExists().toString()
+                + ArtifactInfo.FS
+                + ai.getSignatureExists().toString()
+                + ArtifactInfo.FS
+                + ai.getFileExtension();
+
+        doc.add(FLD_INFO.toField(info));
+
+        doc.add(FLD_GROUP_ID_KW.toField(ai.getGroupId()));
+        doc.add(FLD_ARTIFACT_ID_KW.toField(ai.getArtifactId()));
+        doc.add(FLD_VERSION_KW.toField(ai.getVersion()));
+
+        // V3
+        doc.add(FLD_GROUP_ID.toField(ai.getGroupId()));
+        doc.add(FLD_ARTIFACT_ID.toField(ai.getArtifactId()));
+        doc.add(FLD_VERSION.toField(ai.getVersion()));
+        doc.add(FLD_EXTENSION.toField(ai.getFileExtension()));
+
+        if (ai.getName() != null) {
+            doc.add(FLD_NAME.toField(ai.getName()));
+        }
+
+//        if (ai.getDescription() != null) {
+//            doc.add(FLD_DESCRIPTION.toField(ai.getDescription()));
+//        }
+
+        if (ai.getPackaging() != null) {
+            doc.add(FLD_PACKAGING.toField(ai.getPackaging()));
+        }
+
+        if (ai.getClassifier() != null) {
+            doc.add(FLD_CLASSIFIER.toField(ai.getClassifier()));
+        }
+
+//        if (ai.getSha1() != null) {
+//            doc.add(FLD_SHA1.toField(ai.getSha1()));
+//        }
+    }
+
+    @Override
+    public boolean updateArtifactInfo(Document doc, ArtifactInfo ai) {
+        boolean res = false;
+
+        String uinfo = doc.get(ArtifactInfo.UINFO);
+
+        if (uinfo != null) {
+
+            int start = 0;
+            int end = uinfo.indexOf(FS);
+            ai.setGroupId(uinfo.substring(start, end));
+
+            start = end + 1;
+            end = uinfo.indexOf(FS, start);
+            ai.setArtifactId(uinfo.substring(start, end));
+
+            start = end + 1;
+            end = uinfo.indexOf(FS, start);
+            ai.setVersion(uinfo.substring(start, end));
+
+            start = end + 1;
+            end = uinfo.indexOf(FS, start);
+            if (end == -1) {
+                end = uinfo.length();
+            }
+            ai.setClassifier(ArtifactInfo.renvl(uinfo.substring(start, end)));
+
+            if (end < uinfo.length()) {
+                start = end + 1;
+                end = uinfo.length();
+                ai.setFileExtension(uinfo.substring(start, end));
+            }
+
+            res = true;
+        }
+
+        String info = doc.get(ArtifactInfo.INFO);
+
+        if (info != null) {
+
+            int start = 0;
+            int end = info.indexOf(FS);
+            ai.setPackaging(ArtifactInfo.renvl(info.substring(start, end)));
```

Comment:
Instead of using indexOf+substring to split the `info` string, I would use String#split and work on the resulting String array. From my pov this would improve readability.

Reply:
this is performance critical code, see upstream PR linked above.